src/coreclr-debug/activate.ts

```diff
@@ -7,126 +7,50 @@
 
-class DotnetInfo
-{
-    public Version: string;
-    public OsVersion: string;
-    public RuntimeId: string;
-}
-
-export function activate(context: vscode.ExtensionContext, reporter: TelemetryReporter) {
+export function activate(context: vscode.ExtensionContext, reporter: TelemetryReporter, logger: Logger) {
+    _util = new CoreClrDebugUtil(context.extensionPath, logger);
     _reporter = reporter;
-    _channel = vscode.window.createOutputChannel('coreclr-debug');
-    _util = new CoreClrDebugUtil(context.extensionPath, _channel);
-
-    if (CoreClrDebugUtil.existsSync(_util.installCompleteFilePath())) {
-        console.log('.NET Core Debugger tools already installed');
-        return;
-    }
-
-    checkForDotnetTools().then((dotnetInfo: DotnetInfo) => {
-        let installer = new debugInstall.DebugInstaller(_util);
-        _util.createInstallLog();
-
-        let statusBarMessage = vscode.window.setStatusBarMessage("Downloading and configuring the .NET Core Debugger...");
-
-        let runtimeId: string = '';
-        let installStage: string = "installBegin";
-        let installError: string = '';
-        let moreErrors: string = '';
-
-        getPlatformRuntimeId().then(rid => {
-            runtimeId = rid;
-        }).then(() => {
-            return writeInstallBeginFile();
-        }).then(() => {
-            return installer.install(runtimeId);
-        }).then(() => {
-            installStage = "completeSuccess";
-            statusBarMessage.dispose();
-            vscode.window.setStatusBarMessage('Successfully installed .NET Core Debugger.');
-        }).catch((error: debugInstall.InstallError) => {
-            const viewLogMessage = "View Log";
-            vscode.window.showErrorMessage('Error while installing .NET Core Debugger.', viewLogMessage).then(value => {
-                if (value === viewLogMessage) {
-                    _channel.show(vscode.ViewColumn.Three);
-                }
-            });
-            statusBarMessage.dispose();
-
-            installStage = error.installStage;
-            installError = error.errorMessage;
-            moreErrors = error.hasMoreErrors ? 'true' : 'false';
-        }).then(() => {
-            // log telemetry and delete install begin file
-            logTelemetry('Acquisition', {
-                installStage: installStage,
-                installError: installError,
-                moreErrors: moreErrors,
-                dotnetVersion: dotnetInfo.Version,
-                osVersion: dotnetInfo.OsVersion,
-                osRID: dotnetInfo.RuntimeId
+    _logger = logger;
+
+    if (!CoreClrDebugUtil.existsSync(_util.debugAdapterDir())) {
+        // We have been activated but it looks like our package was not installed. This is bad.
+        logger.appendLine("[ERROR]: C# Extension failed to install the debugger package");
+        showInstallErrorMessage();
+    } else if (!CoreClrDebugUtil.existsSync(_util.installCompleteFilePath())) {
```

Comment:
Since we also have an install complete file for the package manager, I wonder if we should rename some part of this to make the distinction more clear (ex: debugUtil instead of util, or debuggerInstallCompleteFilePath...)